docs/tutorials/services/constrain-motion.md

```diff
@@ -0,0 +1,408 @@
+
+```python {class="line-numbers linkable-line-numbers"}
+# Create a start pose, where the cup starts between the gripper's jaws, on the table
+# Start pose has Z of 90mm to grab partway down the 120mm tall cup
+start_pose = Pose(x=320, y=240, z=90.0+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+start_pose_in_frame = PoseInFrame(reference_frame="world", pose=start_pose)
+
+# Create waypoints to increase efficiency of motion planning
+way1_pose = Pose(x=300, y=240, z=320+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+way1_pose_in_frame = PoseInFrame(reference_frame="world", pose=way1_pose)
+way2_pose = Pose(x=300, y=-240, z=320+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+way2_pose_in_frame = PoseInFrame(reference_frame="world", pose=way2_pose)
+
+# Create a pose where the cup will be set down
+end_pose = Pose(x=300, y=-250, z=90.0+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+end_pose_in_frame = PoseInFrame(reference_frame="world", pose=end_pose)
+```
+
+## Add a motion constraint
+
+To keep the cup upright as the arm moves it from one place on the table to another, create a [linear constraint](../../../services/motion/constraints/#linear-constraint).
+Then, when you tell the robot to move the cup from one upright position to another, the gripper will move linearly and the upright orientation of the cup will be maintained throughout the planned path.
+
+You could try using an [orientation constraint](../../../services/motion/constraints/#orientation-constraint) instead, which would also constrain the orientation.
+However, since this opens up many more options for potential paths, it is much more computationally intensive than the linear constraint.
+
+```python {class="line-numbers linkable-line-numbers"}
+# Create a constraint to keep the cup upright as it moves
+constraints = Constraints(linear_constraint = [LinearConstraint()])
+
+# Move the cup to the end position without hitting the box on the table,
+# and while keeping the cup upright
+await motion_service.move(component_name=my_gripper_resource_name,
+    destination=way1_pose_in_frame, world_state=world_state, constraints=constraints)
+await motion_service.move(component_name=my_gripper_resource_name,
+    destination=way2_pose_in_frame, world_state=world_state, constraints=constraints)
+await motion_service.move(component_name=my_gripper_resource_name,
+    destination=end_pose_in_frame, world_state=world_state, constraints=constraints)
+print(f"At end pose")
+
+# Put down the cup
+await my_gripper.open()
+```
+
+## Full tutorial code
+
+The following code contains everything covered in this tutorial in addition to the connect function, and the resource access code from the last tutorial that you still need here.
+Be sure to change the robot secret and address values to match your actual robot credentials, and change all relevant parameters such as `z_offset` and other dimensions and poses to match your hardware.
```

Comment:
```suggestion
Be sure to change the `<ROBOT SECRET PAYLOAD>` and `<ROBOT ADDRESS>` placeholders shown in the code to match your actual robot credentials, and change all relevant parameters such as `z_offset` and other dimensions and poses to match your hardware.
```

Or similar, just to help the user identify what's being replaced. Maybe no brackets?

Reply:
There are brackets on the previous tutorials--idk